args/physics/colliders/convexcollider.cpp

```diff
@@ -1,76 +1,80 @@
+        PointerEncapsulator < HalfEdgeFace> ARefFace;
 
-        if (PhysicsStatics::FindSeperatingAxisByExtremePointProjection(this, convexCollider, manifold.transformA, manifold.transformB, ARefFace, ARefSeperation))
+        float ARefSeperation;
+        math::vec3 a;
+        if (PhysicsStatics::FindSeperatingAxisByExtremePointProjection(this, convexCollider, manifold.transformB,manifold.transformA,  ARefFace, ARefSeperation, a))
         {
             manifold.isColliding = false;
             return;
         }
 
-        HalfEdgeFace** BRefFace = &convexCollider->GetHalfEdgeFaces().at(0);
+        PointerEncapsulator < HalfEdgeFace> BRefFace;
+      
         float BRefSeperation;
-
-        if (PhysicsStatics::FindSeperatingAxisByExtremePointProjection(convexCollider, this, manifold.transformB, manifold.transformA, BRefFace, BRefSeperation))
+        math::vec3 b;
+        if (PhysicsStatics::FindSeperatingAxisByExtremePointProjection(convexCollider, this, manifold.transformA, manifold.transformB, BRefFace, BRefSeperation, b))
         {
             manifold.isColliding = false;
             return;
         }
 
-        HalfEdgeEdge** edgeRef = &convexCollider->GetHalfEdgeFaces().at(0)->startEdge;
-        HalfEdgeEdge** edgeInc = &this->GetHalfEdgeFaces().at(0)->startEdge;
+        PointerEncapsulator< HalfEdgeEdge> edgeRef;
+        PointerEncapsulator< HalfEdgeEdge> edgeInc;
+
+        //edgeRef.ptr = 
+
         math::vec3 edgeNormal;
         float aToBEdgeSeperation;
 
-        if (PhysicsStatics::FindSeperatingAxisByGaussMapEdgeCheck(this, convexCollider, manifold.transformA, manifold.transformB,
+        if (PhysicsStatics::FindSeperatingAxisByGaussMapEdgeCheck(this, convexCollider, manifold.transformB, manifold.transformA,
             edgeRef, edgeInc, edgeNormal, aToBEdgeSeperation))
         {
             manifold.isColliding = false;
             return;
         }
 
-        //gets the value of edgePositionPtr. HalfEdgeEdge is stored as a pointer to a pointer,
-        //it has a variable edgePositionPtr which is a pointer to a math::vec3.
-        auto edgeposgetter = [](HalfEdgeEdge** p) {
-            return *(*p)->edgePositionPtr;
-        };
+        //assert(*edgeRef);
+
 
-        //--------------------- A Collision has been found, find the most shallow penetration  -------------------------------------//
+        //--------------------- A Collision has been found, find the most shallow penetration  ------------------------------------//
 
         //Get world position and normal of reference faces //
 
-        math::vec3 worldFaceCentroidA = manifold.transformA * math::vec4((*ARefFace)->centroid, 1);
-        math::vec3 worldFaceNormalA = manifold.transformA * math::vec4((*ARefFace)->centroid, 0);
+        math::vec3 worldFaceCentroidA = manifold.transformA * math::vec4((ARefFace.ptr)->centroid, 1);
+        math::vec3 worldFaceNormalA = manifold.transformA * math::vec4((ARefFace.ptr)->normal, 0);
```

Comment:
are these brackets still neccessary ?
`(ARefFace.ptr)->centroid`
Now that this is cleaned up 
`ARefFace.ptr->centroid` should suffice as far as I can tell